src/debug-utils.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { describe, test } from '@jest/globals'
+import algosdk, { AtomicTransactionComposer } from 'algosdk'
+import * as fsSync from 'fs'
+import * as fs from 'fs/promises'
+import * as os from 'os'
+import * as path from 'path'
+import { AVMDebuggerSourceMap, PersistSourceMapInput, persistSourcemaps, simulateAndPersistResponse } from './debug-utils'
+import { algorandFixture } from './testing'
+
+async function fileExists(filePath: string): Promise<boolean> {
+  try {
+    await fs.access(filePath)
+    return true
+  } catch {
+    return false
+  }
+}
+
+describe('debug utils tests', () => {
+  const localnet = algorandFixture()
+  const timeout = 10e6
+
+  beforeAll(localnet.beforeEach)
+
+  test(
+    'build teal sourcemaps',
+    async () => {
+      const { algod } = localnet.context
+      const cwd = await fs.mkdtemp(path.join(os.tmpdir(), 'cwd'))
+
+      const approval = `
+#pragma version 9
+int 1
+`
+      const clear = `
+#pragma version 9
+int 1
+`
+      const sources = [
+        new PersistSourceMapInput(approval, 'cool_app', 'approval.teal'),
+        new PersistSourceMapInput(clear, 'cool_app', 'clear'),
+      ]
+
+      await persistSourcemaps({ sources: sources, projectRoot: cwd, client: algod })
+
+      const rootPath = path.join(cwd, '.algokit', 'sources')
+      const sourcemapFilePath = path.join(rootPath, 'sources.avm.json')
+      const appOutputPath = path.join(rootPath, 'cool_app')
+
+      expect(await fileExists(sourcemapFilePath)).toBeTruthy()
+      expect(await fileExists(path.join(appOutputPath, 'approval.teal'))).toBeTruthy()
+      expect(await fileExists(path.join(appOutputPath, 'approval.teal.tok.map'))).toBeTruthy()
+      expect(await fileExists(path.join(appOutputPath, 'clear.teal'))).toBeTruthy()
+      expect(await fileExists(path.join(appOutputPath, 'clear.teal.tok.map'))).toBeTruthy()
+
+      const result = AVMDebuggerSourceMap.fromDict(JSON.parse(await fs.readFile(sourcemapFilePath, 'utf8')))
+      for (const item of result.txnGroupSources) {
+        item.location = 'dummy'
```

Comment:
Might be worth a comment explaining _why_ we're doing this - I assume it's just to stabilise the output across different machines.